fetch_puzzle_stats.py

```diff
@@ -1,137 +1,170 @@
+        "https://myaccount.nytimes.com/svc/ios/v2/login",
         data={
-            'login': username,
-            'password': password,
+            "login": username,
+            "password": password,
         },
         headers={
-            'User-Agent': 'Crosswords/20191213190708 CFNetwork/1128.0.1 Darwin/19.6.0',
-            'client_id': 'ios.crosswords',
+            "User-Agent": "Crosswords/20191213190708 CFNetwork/1128.0.1 Darwin/19.6.0",
+            "client_id": "ios.crosswords",
         },
     )
     login_resp.raise_for_status()
-    for cookie in login_resp.json()['data']['cookies']:
-        if cookie['name'] == 'NYT-S':
-            return cookie['cipheredValue']
-    raise ValueError('NYT-S cookie not found')
+    for cookie in login_resp.json()["data"]["cookies"]:
+        if cookie["name"] == "NYT-S":
+            return cookie["cipheredValue"]
+    raise ValueError("NYT-S cookie not found")
 
 
-def get_puzzle_stats(date, cookie):
+def get_v3_puzzle_overview(puzzle_type, start_date, end_date, cookie):
+    payload = {
+        "publish_type": puzzle_type,
+        "sort_order": "asc",
+        "sort_by": "print_date",
+        "date_start": start_date.strftime("%Y-%m-%d"),
+        "date_end": end_date.strftime("%Y-%m-%d"),
+    }
+
+    overview_resp = requests.get(PUZZLE_INFO, params=payload, cookies={"NYT-S": cookie})
+
+    overview_resp.raise_for_status()
+    puzzle_info = overview_resp.json().get("results")
+    return puzzle_info
+
+
+def get_v3_puzzle_detail(puzzle_id, cookie):
     puzzle_resp = requests.get(
-        PUZZLE_INFO.format(date=date),
-        cookies={
-            'NYT-S': cookie,
-        },
+        f"{PUZZLE_DETAIL}/{puzzle_id}.json", cookies={"NYT-S": cookie}
     )
+
     puzzle_resp.raise_for_status()
-    puzzle_date = datetime.strptime(date, DATE_FORMAT)
-    puzzle_info = puzzle_resp.json().get('results')[0]
-    solve_resp = requests.get(
-        SOLVE_INFO.format(game_id=puzzle_info['puzzle_id']),
-        cookies={
-            'NYT-S': cookie,
-        },
-    )
-    solve_resp.raise_for_status()
-    solve_info = solve_resp.json().get('results')
-
-    solved = solve_info.get('solved', False)
-    checked = 'firstChecked' in solve_info
-    revealed = 'firstRevealed' in solve_info
-    solve_date = datetime.fromtimestamp(solve_info.get('firstSolved', 0))
-    # A puzzle is streak eligible if they didn't cheat and they solved it
-    # before midnight PST (assume 8 hour offset for now, no DST)
-    streak_eligible = solved and not checked and not revealed and (
-        solve_date <= puzzle_date + timedelta(days=1) + timedelta(hours=8))
-
-    return {
-        'elapsed_seconds': solve_info.get('timeElapsed', 0),
-        'solved': int(solved),
-        'checked': int(checked),
-        'revealed': int(revealed),
-        'streak_eligible': int(streak_eligible),
-    }
+    puzzle_detail = puzzle_resp.json()["calcs"]
 
+    # crude rate limiting
+    time.sleep(0.25)
```

Comment:
Did you notice that rate limiting was necessary? I hadn't encountered any rate limiting on NYT APIs in the past but maybe the new APIs have it? This would make it take over a minute just to get 1 year's worth of solves though so would rather avoid if possible

Reply:
I had one Mysterious Slowdown and didn't have my debugger running, so I wasn't able to check out the response codes :( 

So this is a superstitious fix and I'm happy to let it go